testsuite/tests/ui/oas/test_active_docs_v3.py

```diff
@@ -0,0 +1,136 @@
+import pytest
+from threescale_api.resources import Service
+
+from testsuite import rawobj
+from testsuite.ui.views.admin.audience.developer_portal import ActiveDocsNewView
+from testsuite.ui.views.admin.product.active_docs import ActiveDocsDetailView
+from testsuite.utils import blame
+
+
+@pytest.fixture(scope="module")
+def service_settings2(request):
+    """settings for second service"""
+    return {"name": blame(request, "svc"),
+            "backend_version": Service.AUTH_APP_ID_KEY}
+
+
+@pytest.fixture(scope="module")
+def service2(backends_mapping, custom_service, service_settings2, service_proxy_settings, lifecycle_hooks):
+    """Second service to test with"""
+    return custom_service(service_settings2, service_proxy_settings, backends_mapping, hooks=lifecycle_hooks)
+
+
+@pytest.fixture(scope="module")
+def active_doc2(request, service2, oas3_body, custom_active_doc):
+    """Active doc. bound to second service."""
+    return custom_active_doc(rawobj.ActiveDoc(blame(request, "activedoc"), oas3_body, service=service2),
+                             autoclean=False)
+
+
+@pytest.fixture(scope="module")
+def oas3_spec():
+    """
+    OAS3 Active doc spec with 3scale `user key` parameter
+    """
+    user_key = {
+        "name": "user_key",
+        "description": "Your access API Key",
+        "type": "string",
+        "in": "query",
+        "x-data-threescale-name": "user_keys",
+        "required": True
+    }
+    oas_spec = resources.files("testsuite.resources.oas3").joinpath("petstore-expanded.json").read_text()
+    json_spec = json.loads(oas_spec)
+    json_spec['paths']['/pets']['get']['parameters'].append(user_key)
+    oas_spec = json.dumps(json_spec)
+    return oas_spec
+
+
+@pytest.fixture(scope="module")
+def prod_client(prod_client):
+    """
+    Production client so tests can send request to service production endpoint
+    """
+    client = prod_client()
+    response = client.get('/get')
+    assert response.status_code == 200
+    return client
+
+
+# pylint: disable=unused-argument
+@pytest.fixture(scope='module')
+def ui_active_doc(custom_admin_login, request, navigator, service, oas3_spec):
+    """Active doc. bound to service created via UI"""
+    custom_admin_login()
+    name = blame(request, "active_doc_v3")
+    system_name = blame(request, "system_name")
+    edit = navigator.navigate(ActiveDocsNewView)
+    edit.create_spec(name=name,
+                     sys_name=system_name,
+                     description="Active docs V3",
+                     service=service,
+                     oas_spec=oas3_spec,
+                     publish_option=True,
+                     skip_validation_option=True)
+    return service.active_docs.list()[0]
+
+
+# pylint: disable=unused-argument, too-many-arguments
+@pytest.mark.disruptive  # test should be mark as disruptive because of production gateway redeploy
+@pytest.mark.issue("https://issues.redhat.com/browse/THREESCALE-770",
```

Comment:
Issue 770 includes a service discovery feature. It should be covered with the test or removed from this test.